app/src/org/commcare/activities/CommCareActivity.java

```diff
@@ -189,6 +196,18 @@ private void persistManagedUiState() {
         }
     }
 
+    @Override
+    public void onPostCreate(@Nullable Bundle savedInstanceState) {
+        super.onPostCreate(savedInstanceState);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.VANILLA_ICE_CREAM) {
+            AndroidUtil.attachWindowInsetsListener(this, getRootViewId());
+        }
+    }
+
+    public int getRootViewId() {

```

Comment:
1. this should be an abstract method as activities must implement this. 
2. On Connect, a single activity host multiple fragments with their own views. Is there no generalised way to get a root view of the current screen instead of assigning IDs to the xml layouts ?